Assets/Scripts/NPCs/Patroller.cs

```diff
@@ -76,30 +81,22 @@ void GotoNextPoint()
         destinationCount++;
     }
 
-    public void ChaseTarget()
-    {
-        Vector3 targetPosition = targetTransform.position;
-        Vector3 dirToTarget = transform.position - targetPosition;
-        Vector3 newPos = transform.position - dirToTarget;
-
-        agent.SetDestination(newPos);
-    }
-
     void Update()
     {
         if (agent.isOnNavMesh && !agent.pathPending && agent.remainingDistance < 0.5f)
         {
             switch (currentState)
             {
-                case NurseStates.Patrolling:
+                case ActionStates.Patrolling:
                     // Choose the next destination point when the agent gets close to the current one.
-                    GotoNextPoint();
+                    if (WaypointPauseComplete())
+                    {
+                        GotoNextPoint();
+                    }
                     break;
-                case NurseStates.Chasing:
+                case ActionStates.Chasing:
                     ChaseTarget();
                     break;
-                default:
```

Comment:
Switch statement should probably have a default statement right? Maybe we should also log something if this happens?

Reply:
Throwing an exception can be a good idea like you did below.

Reply:
Will do, I'll add a failsafe switch case that throws an exception.